### PR TITLE
Update to remove MalwareBytes as recommended software

### PR DESCRIPTION
**Pull request by sadmoonphoenix**
MalwareBytes is not good, and should not be recommended. 

They have since stopped benchmarking it (since it's that bad), but here's a benchmark. https://www.av-comparatives.org/tests/malware-protection-test-september-2022/

As you can see, it has lower detection rates, and higher false positive rates. 

I have also not been able to reproduce the "issues" so called caused by BitDefender, and have likely been fixed. (And even so, it would be trivial to set an exception for Minecraft.) If someone is alone to reproduce this behavior on the newest version of BitDefender, I am happy to back down. 

The bottom line is: we should not be promoting bad software based on what might have been true in the past.

### CODE REVIEW

**Comment by randomairborne** [comment]
Perhaps we shouldn't promote third-party software at all. Windows Defender has improved quite significantly in the last years.
I believe we initially recommended malwarebytes because they were really good at detecting certain malware that *does* cause issues with minecraft with their ADWCleaner tool? Someone else would have to confirm that. A choice of general AV is definitely outside our scope.

**Comment by sadmoonphoenix** [comment]
In that case, we shouldn't list AVs as incompatible software. They're bound to cause issues, but those issues are swiftly fixed. It's misleading.

**Comment by sadmoonphoenix** [comment]
But the truth of the matter is, the damage is done. If we don't update with an alternative, people will continue to recommend MalwareBytes, regardless if it's still on the resource or not.

**Comment by TheUnusualGuy** [comment]
> In that case, we shouldn't list AVs as incompatible software. They're bound to cause issues, but those issues are swiftly fixed. It's misleading.

These AVs are listed as incompatible because they come bundled with other stuff that slows down the game or actually does cause issues.

**Comment by Whomsky** [comment]
> In that case, we shouldn't list AVs as incompatible software. They're bound to cause issues, but those issues are swiftly fixed. It's misleading.

The thing is, the AVs listed are literally that. Incompatible by default. So they're listed as incompatible software. Why would we change that?

Depending on the AV, we actually provide instructions (usually in factoids) on how to add an exception to fix the issue, but some AVs interfere so bad to the point where we just recommend uninstalling them, as they're generally bad AVs anyways. (Think McAfee)

That being said, it's impossible to keep up with all changes made to AVs, so some of the listed AVs may work fine with Minecraft nowadays. This list is more than 10 years old now and we try our best to maintain it. If you spot any wrong info, tell us in a PR and we can test it together.

**Comment by randomairborne** [comment]
I would like to add that the vital advantage of defender is not actually that it's much better then the alternatives at its job, it's about even with them. It's that it's pre-installed, and it's "just as good" without a lot of bloat and subscriptions.

**Comment by sadmoonphoenix** [comment]
> > In that case, we shouldn't list AVs as incompatible software. They're bound to cause issues, but those issues are swiftly fixed. It's misleading.
> 
> The thing is, the AVs listed are literally that. Incompatible by default. So they're listed as incompatible software. Why would we change that?
> 
> Depending on the AV, we actually provide instructions (usually in factoids) on how to add an exception to fix the issue, but some AVs interfere so bad to the point where we just recommend uninstalling them, as they're generally bad AVs anyways. (Think McAfee)
> 
> That being said, it's impossible to keep up with all changes made to AVs, so some of the listed AVs may work fine with Minecraft nowadays. This list is more than 10 years old now and we try our best to maintain it. If you spot any wrong info, tell us in a PR and we can test it together.

That's what I just did. I haven't experienced any issues with BitDefender, and Minecraft, neither have several other people I've talked to, and it's objectively better than MalwareBytes at the thing it's supposed to do.

**Comment by sadmoonphoenix** [comment]
> I would like to add that the vital advantage of defender is not actually that it's much better then the alternatives at its job, it's about even with them. It's that it's pre-installed, and it's "just as good" without a lot of bloat and subscriptions.

It's not really just as good. 


https://www.av-comparatives.org/tests/malware-protection-test-september-2023/
https://www.av-comparatives.org/tests/real-world-protection-test-july-october-2023/

**Comment by randomairborne** [comment]
Defender's protection online is within .05% of Kaspersky and Bitdefender, for $40 and $80 a year, respectively, for the *cheapest plan*. It does falter in the offline category, but, uh... this is for a website. I don't think most people can justify that, and recommending any AV at this point seems silly. I agree that recommendations should be removed, but certainly not in favor of any other AV.

**Comment by sadmoonphoenix** [comment]
> Defender's protection online is within .05% of Kaspersky and Bitdefender, for $40 and $80 a year, respectively, for the *cheapest plan*. It does falter in the offline category, but, uh... this is for a website. I don't think most people can justify that, and recommending any AV at this point seems silly. I agree that recommendations should be removed, but certainly not in favor of any other third-party AV.

Not sure what you're missing, but Kaspersky's and BitDefender's cheapest plan is $0. The same is true with MalwareBytes, which also offers a paid subscription. There is no change here.

**Comment by sadmoonphoenix** [comment]
The difference between BitDefender's protection and MalwareBytes's protection is 17 infections.

**Comment by randomairborne** [comment]
I visited Kaspersky and Bitdefender's websites, and I found no mention of a free plan. It might be hidden away somewhere, I just didn't check. Recommending malwarebytes is pretty clearly the wrong option here, but I feel recommending any antivirus is not a good decision.

**Comment by sadmoonphoenix** [comment]
> I visited Kaspersky and Bitdefender's websites, and I found no mention of a free plan. It might be hidden away somewhere, I just didn't check. Recommending malwarebytes is pretty clearly the wrong option here, but I feel recommending any antivirus is not a good decision.

Why do you feel it's a bad decision? You've been perfectly fine doing it up until 30 minutes ago when this PR was created. I don't understand what's changed. 

You're right that the BitDefender page hides the free version. I will update the PR. 

I am too lazy to find the Kaspersky link, and frankly I'm going to leave it off the table due to privacy concerns associated with the company.

**Comment by randomairborne** [comment]
Because putting it there in the first place was probably a bad decision.

**Comment by sadmoonphoenix** [comment]
> Because putting it there in the first place was probably a bad decision.

Not recommending an antivirus is recommending Defender, which I think we established as not good enough.

**Comment by randomairborne** [comment]
Essentially my only argument here has been "yes, it is good enough" 😅

**Comment by sadmoonphoenix** [comment]
It's not. 

https://www.av-comparatives.org/tests/real-world-protection-test-july-october-2023/
https://www.av-comparatives.org/tests/malware-protection-test-september-2023/

**Comment by randomairborne** [comment]
Saying "It's not" and pointing at these same two links, when I did already read over them and [make an argument](https://github.com/MinecraftHopper/MinecraftHopper/pull/149#issuecomment-1926238718) for why it is based on them, is not very helpful.

**Comment by sadmoonphoenix** [comment]
My argument is that measuring protection rates as a percentage rate is confusing and misleading. But fine, maybe we can come to a compromise. Hold on.

**Comment by sadmoonphoenix** [comment]
> Saying "It's not" and pointing at these same two links, when I did already read over them and [make an argument](https://github.com/MinecraftHopper/MinecraftHopper/pull/149#issuecomment-1926238718) for why it is based on them, is not very helpful.

Your "0.05%" is a small number, but when represented as absolutes, that means 3 more missed detections in the synthetic tests and 4 more missed detections in the real world tests. That's far from insignificant. Stop using purposely misleading numbers.

**Comment by randomairborne** [comment]
I personally would not consider those 5 detections out of literally thousands of tests to be statistically significant. While they may to you show a better antivirus, the average joe wouldn't care. 5 in 10,000 is... well, it's not that big of a number. As much as 5 virus infections probably sounds big, in this sample size it really doesn't seem significant. In addition, updating this frequently with whatever good third party antivirus exists is (as you can see) not a realistic proposition. As long as windows is big, microsoft will probably support defender, and it will likely not only be good enough now, but for a long time in the future.

**Comment by sadmoonphoenix** [comment]
10,000 is a big sample size, sure. Regardless, similar results are found in the real world testing, so I'd like to consider that argument void. If you have a reason I shouldn't, please tell me. 

I'd also like you to reason why 
```
Use BitDefender unless you don't want to. Also do research if you're curious about the difference between BitDefender and Windows Defender
```
is unsatisfactory to you. I get what I want - recommending a good AV, and you get what you want - [...].

**Comment by sadmoonphoenix** [comment]
> I personally would not consider those 5 detections out of literally thousands of tests to be statistically significant. While they may to you show a better antivirus, the average joe wouldn't care. 5 in 10,000 is... well, it's not that big of a number. As much as 5 virus infections probably sounds big, in this sample size it really doesn't seem significant. In addition, updating this frequently with whatever good third party antivirus exists is (as you can see) not a realistic proposition. As long as windows is big, microsoft will probably support defender, and it will likely not only be good enough now, but for a long time in the future. 

Windows will support Defender, sure. I fail to see how this carries over to BitDefender though. They'll also support their product. They make money off of it.

**Review by TheUnusualGuy — COMMENTED** [review]
I would recommend not approving this.

**Comment by sadmoonphoenix** [comment]
@TheUnusualGuy why? I fail to see any reason supplied why this wouldn't benefit literally everyone using this as a resource. MalwareBytes is not a reliable AV. It is literally one sentence changed, not a massive code rebase.

**Comment by urielsalis** [comment]
Lets take this to qa-with-helpers

**Comment by sadmoonphoenix** [comment]
> Lets take this to qa-with-helpers

Okay. I'm planning to poll some people who use BitDefender. I'll get back to you guys with the results.

**Comment by sadmoonphoenix** [comment]
> I personally would not consider those 5 detections out of literally thousands of tests to be statistically significant. While they may to you show a better antivirus, the average joe wouldn't care. 5 in 10,000 is... well, it's not that big of a number. As much as 5 virus infections probably sounds big, in this sample size it really doesn't seem significant. In addition, updating this frequently with whatever good third party antivirus exists is (as you can see) not a realistic proposition. As long as windows is big, microsoft will probably support defender, and it will likely not only be good enough now, but for a long time in the future.

Also, not sure if you noticed, but further down on the Hopper website, it literally recommends using an AV to remove malware, likely because Defender didn't detect it. So, obviously, it isn't enough. BitDefender is plain better, and I don't understand why we're so reluctant to change a resource to be more accurate.

**Comment by urielsalis** [comment]
This is an old recommendation when Malwarebytes was one of the few good free antiviruses, and its been there since the hopper.minecraft.net article. 

I think we should remove this line entirely and not recommend anything inside this article

File: _help/known-incompatible-software/index.html [hunk 1]
```diff
@@ -174,7 +167,7 @@
       Not Detected by ADWCleaner
       Presents like a VPN, Can cause download issues and other issues with the game
 
-      Download [Malwarebytes](http://www.malwarebytes.org/) run a **Scan**, then **Quarentine**. Restart when Malwarebytes tells you to, then try running the game again.
+      Download [BitDefender Free](https://www.bitdefender.com/solutions/free.html) run a **System Scan**. Restart when BitDefender tells you to, then try running the game again.
```
Comment (urielsalis) [inline _help/known-incompatible-software/index.html:170]: The reason we recommend malwarebytes here is that has a specific detection and removal for this, instead of following a series of step

Comment (urielsalis) [inline _help/known-incompatible-software/index.html:170]: The reason we recommend malwarebytes here is that has a specific detection and removal for this, instead of following a series of step

**Review by urielsalis — CHANGES_REQUESTED** [review]

**Comment by sadmoonphoenix** [comment]
> This is an old recommendation when Malwarebytes was one of the few good free antiviruses, and its been there since the hopper.minecraft.net article. 
> 
> I think we should remove this line entirely and not recommend anything inside this article 

That doesn't work. Some of your incompatible software listed needs anti malware to be removed. Why not just keep the BitDefender rec. I don't understand what's wrong with recommending it.

File: _help/known-incompatible-software/index.html (hunk 1 above)
Comment (sadmoonphoenix) [reply _help/known-incompatible-software/index.html:170]: The chances that MalwareBytes detects this and BitDefender doesn't is next to zero. Next stupid argument please!

Comment (urielsalis) [reply _help/known-incompatible-software/index.html:170]: Before suggesting a change you would need to test it, do you have access to that software to verify bitdefender removes it corectly?

Comment (sadmoonphoenix) [reply _help/known-incompatible-software/index.html:170]: I'm sure I can find a strain somewhere. If someone has one, please let me know.

**Comment by urielsalis** [comment]
> > This is an old recommendation when Malwarebytes was one of the few good free antiviruses, and its been there since the hopper.minecraft.net article.
> > I think we should remove this line entirely and not recommend anything inside this article
> 
> That doesn't work. Some of your incompatible software listed needs anti malware to be removed. Why not just keep the BitDefender rec. I don't understand what's wrong with recommending it.

If you want to suggest changing to BitDefender, we would need to test it with all software that is included in the list as a minimum to see if it removes it the same as malwarebytes.

Its easier to use something we know works even if its not the best, than something new.

If there is anything remaining, helpers can always suggest next steps

File: _help/known-incompatible-software/index.html (hunk 1 above)
Comment (sadmoonphoenix) [reply _help/known-incompatible-software/index.html:170]: If you don't have a sample, then I can't. IDM is a legitimate program. What you likely found is some Trojan disguising itself. In this case, BitDefender is better, since it has a higher detection rate.

**Comment by LordRalex** [comment]
We will not recommend BitDefender.

Your entire starting argument is flawed. That site does *NOT* say that MalwareBytes is bad. They even gave it a damn recommendation in Feb 2023, months after your own tests you keep citing.
https://www.av-comparatives.org/av-comparatives-awards-2022-for-malwarebytes/

BitDefender caused us enough headaches in the past, we are not going to immediately recommend it.

Your entire argument stems off of "it missed 5 tests!!!!". Cool. I don't care. No AV is perfect. So, based on your own arguments you keep using, no AV should be recommend.

And insulting us is a good way to deny this.

This is done. We will not accept your PR.
